ViewModel/HyvaGridViewModel.php

```diff
@@ -143,17 +143,19 @@ private function getGridSourceModel(): HyvaGridSourceInterface
         return $this->memoizedGridSource;
     }
 
-    public function getRows(SearchCriteriaInterface $searchCriteria = null): array
+    public function getRows(): array
     {
-        if( is_null($searchCriteria) ) {
-            $searchCriteria = $this->getNavigation()->getSearchCriteria();
-        }
         if (!isset($this->currentData)) {
-            $this->currentData = map([$this, 'buildRow'], $this->getGridSourceModel()->getRecords($searchCriteria));
+            $this->currentData = $this->getRowsForSearchCriteria($this->getNavigation()->getSearchCriteria());
```

Comment:
Entferne bitte die Eigenschaft `currentData`. Die geladenen Daten werden schon in dem GridSource memoizied (mit dem besprochenen Fix das sie in Abhängigkeit von der SearchCriteria memoized werden sollten).